api/envoy/config/common/mutation_rules/v3/mutation_rules.proto

```diff
@@ -90,6 +91,12 @@ message HeaderMutationRules {
 // The HeaderMutation structure specifies an action that may be taken on HTTP
 // headers.
 message HeaderMutation {
+  message RemoveOnMatch {
```

Comment:
Do you anticipate more fields being added to this message?  If not, should StringMatcher just be inlined into the `oneof action`?

Reply:
Yeah. Now, only header key will be used for matching. I think it's possible to provide more match conditions in the future.